component/otelcol/config_tls.go

```diff
@@ -70,17 +70,31 @@ func (args *TLSSetting) Convert() *otelconfigtls.TLSSetting {
 		return nil
 	}
 
-	return &otelconfigtls.TLSSetting{
-		CAPem:          configopaque.String(args.CA),
+	res := &otelconfigtls.TLSSetting{
 		CAFile:         args.CAFile,
-		CertPem:        configopaque.String(args.Cert),
 		CertFile:       args.CertFile,
-		KeyPem:         configopaque.String(string(args.Key)),
 		KeyFile:        args.KeyFile,
 		MinVersion:     args.MinVersion,
 		MaxVersion:     args.MaxVersion,
 		ReloadInterval: args.ReloadInterval,
 	}
+
+	// Collector sets the pem attributes to nil by default,
+	// so let's match this.
```

Comment:
nit: nil or ""?

Reply:
Sorry, I'm going to revert this change. This used to be a byte array which would default to nil, but then Otel changed it upstream to a string which defaults to "". The unit tests still pass ok when this change is reverted, so I think it's not needed. Thanks for spotting it!